KenticoCloud.Delivery.Rx.Tests/DeliveryObservableFactoryTests.cs

```diff
@@ -0,0 +1,295 @@
+{
+    public class DeliveryObservableFactoryTests
+    {
+        private const string BEVERAGES_IDENTIFIER = "coffee_beverages_explained";
+        readonly string guid = string.Empty;
+        readonly string baseUrl = string.Empty;
+        readonly MockHttpMessageHandler mockHttp;
+
+        public DeliveryObservableFactoryTests()
+        {
+            guid = Guid.NewGuid().ToString();
+            baseUrl = $"https://deliver.kenticocloud.com/{guid}";
+            mockHttp = new MockHttpMessageHandler();
+        }
+
+        [Fact]
+        public async void ItemJsonRetrieved()
+        {
+            var observable = DeliveryObservableFactory.ItemJson(GetDeliveryClient(MockItem), BEVERAGES_IDENTIFIER, "language=es-ES");
+            var itemJson = await observable.FirstOrDefaultAsync();
+
+            Assert.Single(observable.ToEnumerable());
+            Assert.NotNull(itemJson);
+        }
+
+        [Fact]
+        public void ItemsJsonRetrieved()
+        {
+            var observable = DeliveryObservableFactory.ItemsJson(GetDeliveryClient(MockItems), "limit=2", "skip=1");
+            var itemsJson = observable.ToEnumerable().ToList();
+
+            Assert.NotEmpty(itemsJson);
+            Assert.InRange(itemsJson[0]["items"].Count(), 2, int.MaxValue);
+        }
+
+        [Fact]
+        public async void ContentItemRetrieved()
+        {
+            var observable = DeliveryObservableFactory.Item(GetDeliveryClient(MockItem), BEVERAGES_IDENTIFIER, new LanguageParameter("es-ES"));
+            var item = await observable.FirstOrDefaultAsync();
+
+            Assert.NotNull(item);
+            AssertItemPropertiesNotNull(item);
+        }
+
+        [Fact]
+        public async void TypedItemRetrieved()
+        {
+            var observable = DeliveryObservableFactory.Item<Article>(GetDeliveryClient(MockItem), BEVERAGES_IDENTIFIER, new LanguageParameter("es-ES"));
+            var item = await observable.FirstOrDefaultAsync();
+
+            Assert.NotNull(item);
+            AssertArticlePropertiesNotNull(item);
+        }
+
+        [Fact]
+        public async void RuntimeTypedItemRetrieved()
+        {
+            var observable = DeliveryObservableFactory.Item<object>(GetDeliveryClient(MockItem), BEVERAGES_IDENTIFIER, new LanguageParameter("es-ES"));
+            var item = await observable.FirstOrDefaultAsync();
+
+            Assert.IsType<Article>(item);
+            Assert.NotNull(item);
+            AssertArticlePropertiesNotNull((Article)item);
+        }
+
+        [Fact]
+        public void ContentItemsRetrieved()
+        {
+            var observable = DeliveryObservableFactory.Items(GetDeliveryClient(MockItems), new LimitParameter(2), new SkipParameter(1));
+            var items = observable.ToEnumerable().ToList();
+
+            Assert.NotEmpty(items);
+            Assert.InRange(items.Count, 2, int.MaxValue);
+            items.ForEach(i => AssertItemPropertiesNotNull(i));
```

Comment:
use semantic variables please.

Isn't there any other solution for forearch:
I have found: Assert.All(items, item => AssertItemPropertiesNotNull(item));